cpp/src/gandiva/precompiled/time.cc

```diff
@@ -944,6 +945,92 @@ gdv_int64 castBIGINT_daytimeinterval(gdv_day_time_interval in) {
          extractDay_daytimeinterval(in) * MILLIS_IN_DAY;
 }
 
+static const char* PATTERN_MONTH[] = {"MONTH", "MON", "MM"};
+static const int PARTTERN_MONTH_LEN[] = {5, 3, 2};
+
+static const char* PATTERN_YEAR[] = {"YEAR", "YYYY", "YY"};
+static const int PARTTERN_YEAR_LEN[] = {4, 4, 2};
+
+FORCE_INLINE
+const char* date_trunc_timestamp_utf8(int64_t context, gdv_int64 date,
+                                      const char* pattern_name, gdv_int32 pattern_length,
+                                      gdv_int32* out_len) {
+  if (pattern_length <= 1) {
+    gdv_fn_context_set_error_msg(
+        context, "The parameter pattern_length is not contain a valid value");
+    *out_len = 0;
+    return "";
+  }
+
+  gdv_timestamp dateMax = castTIMESTAMP_utf8(context, "9999-12-31 23:59:59", 19);
```

Comment:
As this value is a constant, I think you should move it outside the function and store it as a static variable.